helpers/class.SessionManager.php

```diff
@@ -111,13 +111,22 @@ public function getTest() {
      */
     protected function instantiateAssessmentTestSession(AssessmentTest $test, Route $route) {
         $config = \common_ext_ExtensionsManager::singleton()->getExtensionById('taoQtiTest')->getConfig('testRunner');
+        
+        // Test Session class instantiation, depending on configuration.
         if (!isset($config) || !isset($config['test-session'])) {
             $className = self::DEFAULT_TEST_SESSION;
             \common_Logger::w("Missing configuration for TestRunner session class, using '${className}' by default!");
         } else {
             $className = $config['test-session'];
         }
-        return new $className($test, $this, $route, $this->getResultServer(), $this->getTest());
+        
+        $assessmentTestSession = new $className($test, $this, $route, $this->getResultServer(), $this->getTest());
+        
+        $assessmentTestSession->setForceBranching($config['force-branchrules']);
+        $assessmentTestSession->setForcePreconditions($config['force-preconditions']);
+        $assessmentTestSession->setPathTracking($config['path-tracking']);
```

Comment:
Can lead to issue if the config does not contains the attributes. It should be better  to check before with some `isset()`